src/hooks/api/useFetch.js

```diff
@@ -0,0 +1,131 @@
+import { useState, useEffect, useRef } from "react";
+import axios from "axios";
+import { baseApiUrl } from "@/config/environment";
+import { logInfo } from "@/utils/logging";
+
+const useFetch = (
+  initialRoute,
+  method = "GET",
+  body = null,
+  customHeaders = {},
+  onReceived,
```

Comment:
In case you want to provide a default for the onReceived callback, it’s a good to set it as an empty function by default: `onReceived = () => {}` 😊

Reply:
Thank you very much for your work, effort and patience. Your feedback was great, the code turned out excellent thanks to you!